src/App.tsx

```diff
@@ -8,53 +8,151 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { useEffect, useState } from 'react';
+import { User } from './types/User';
+import { getUserPosts } from './api/ClientFunctions';
+import { Post } from './types/Post';
 
-export const App = () => (
-  <main className="section">
-    <div className="container">
-      <div className="tile is-ancestor">
-        <div className="tile is-parent">
-          <div className="tile is-child box is-success">
-            <div className="block">
-              <UserSelector />
-            </div>
+export const App = () => {
+  // #region states
 
-            <div className="block" data-cy="MainContent">
-              <p data-cy="NoSelectedUser">No user selected</p>
+  const [currUser, setCurrUser] = useState<User | null>(null);
+  const [isDdActive, setIsDdActive] = useState(false);
+  const [userPosts, setUserPosts] = useState<Post[]>([]);
+  const [loadingPostsError, setLoadingPostsError] = useState(false);
+  const [arePostsLoading, setArePostsLoading] = useState(false);
+  const [isNoPosts, setIsNoPosts] = useState(false);
+  const [openedPost, setOpenedPost] = useState<Post | null>(null);
+  const [prevUser, setPrevUser] = useState<User | null>(null);
+  const [doesFormExist, setDoesFormExist] = useState(false);
 
-              <Loader />
+  // #endregion
+  // #region handlers
 
-              <div
-                className="notification is-danger"
-                data-cy="PostsLoadingError"
-              >
-                Something went wrong!
-              </div>
+  const sectionOnCLickHandler = () => {
```

Comment:
Typo in function name: `sectionOnCLickHandler` should be `sectionOnClickHandler` to maintain consistency and readability.